Libplanet/Blockchain/BlockChain.cs

```diff
@@ -468,6 +511,30 @@ internal BlockLocator GetBlockLocator(int threshold = 10)
         // we need to add a synchronization mechanism to handle this correctly.
         internal void Swap(BlockChain<T> other)
         {
+            // Finds the branch point.
+            Block<T> topmostCommon = null;
+            for (long i = Math.Min(other.Tip.Index, Tip.Index); i >= 0; i--)
```

Comment:
We can use `Block<T>.PreviousHash` to avoid index lookup.

Reply:
Fixed: 9db8f48c4dd539b17559cf3723a9703f28cbeedd.